src/io/react-native/react-native-worker.ts

```diff
@@ -193,10 +195,14 @@ async function makeIo(clientIo: ClientIo): Promise<EdgeIo> {
       if (doFetchCors) {
         try {
           const response = await fetchCorsProxy(uri, opts)
+          if (response.status === 418) {
+            throw new Error(PROXY_ERROR)
+          }
           hostnameProxyWhitelist.add(hostname)
           return response
         } catch (error) {
-          if (errorToThrow == null) errorToThrow = error
+          if (errorToThrow == null && String(error) !== PROXY_ERROR)
```

Comment:
Here you can use `asMaybeNetworkError` to see if the error thrown is from the CORS proxy.